source/initial_conditions/adiabatic.cc

```diff
@@ -58,6 +52,21 @@ namespace aspect
                                                   :
                                                   adiabatic_surface_temperature;
 
+      // then, get the temperature at the top and bottom boundary of the model
+      // if no boundary temperature is prescribed simply use the adiabatic
+      const double T_surface = (&this->get_boundary_temperature() != 0)
+                               ?
+                                   this->get_boundary_temperature().minimal_temperature(
+                                       this->get_fixed_temperature_boundary_indicators())
+                               :
+                                   adiabatic_surface_temperature;
+      const double T_bottom = (&this->get_boundary_temperature() != 0)
+                              ?
+                                  this->get_boundary_temperature().maximal_temperature(
+                                      this->get_fixed_temperature_boundary_indicators())
+                              :
+                                  adiabatic_surface_temperature;
+
```

Comment:
Yes, I think this is the best we can do. It's also probably going to be what the user wants, given that it doesn't make much sense to ask for the adiabatic temperature profile if the top is hotter than the bottom.

Please update the comment and merge yourself.